packages/labs/src/frameworks/react/create-component.ts

```diff
@@ -0,0 +1,237 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as ReactModule from 'react';
+
+const reservedReactProperties = new Set([
+  'children',
+  'localName',
+  'ref',
+  // TODO(sorvell): why are the properties below included?
+  'style',
+  'className',
+]);
+
+const listenedEvents: WeakMap<
+  Element,
+  Map<string, EventListenerObject>
+> = new WeakMap();
+
+const setEvent = (
+  node: Element,
+  event: string,
+  value: (event?: Event) => void
+) => {
+  let events = listenedEvents.get(node);
+  if (events === undefined) {
+    listenedEvents.set(node, (events = new Map()));
+  }
+  let handler = events.get(event);
+  if (value !== undefined) {
+    // If necessary, add listener and track handler
+    if (handler === undefined) {
+      events.set(event, (handler = {handleEvent: value}));
+      node.addEventListener(event, handler);
+      // Otherwise just update the listener with new value
+    } else {
+      handler.handleEvent = value;
+    }
+    // Remove listener if one exists and value is undefined
+  } else if (handler !== undefined) {
+    events.delete(event);
+    node.removeEventListener(event, handler);
+  }
+};
+
+const setProperty = <E extends Element, T>(
+  node: E,
+  name: string,
+  value: unknown,
+  _old: unknown,
+  events?: StringValued<T>
+) => {
+  if (reservedReactProperties.has(name)) {
+    return;
```

Comment:
Shouldn't reserved properties never reach this point? The need to be passed on to the underlying `createElement` call, so they shouldn't be filtered.

Also, should we warn at `createComponent()` time if the element class contains reserved properties?

Reply:
Fixed.